### PR TITLE
Fix bug where a discarded browser tab could re-init an app to an obso…

### PR DESCRIPTION
**Pull request by ghsolomon**
…lete (cached) version.

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,10 @@
   type, a union of `CompoundFilter | FieldFilter` - the two concrete filter implementations
   supported by this control.
 
+### 🐞 Bug Fixes
+
+* Fixed bug where a discarded browser tab could re-init an app to an obsolete (cached) version.
+
```
Comment (amcclain) [inline CHANGELOG.md:24]: Need to move to new SNAP release before merging

**Comment by lbwexler** [comment]
Hi Greg:   Love this, but can we try something like this API?

```
   const url = new URL(location.href);
        url.searchParams.set('xhCacheBuster', Date.now().toString());
        document.location.assign(url);
```
        
Seems to work well and would allow us to:
- Avoid problems when there is a '#'
- Allow us to check the date of any existing value and avoid getting into an endless loop.
- just set a single value in the URL

Also -- should we use this generally wherever we do reload()

**Comment by lbwexler** [comment]
that is, we could just build this in to XH.reloadApp() and maybe avoid other caching issues as well.

**Comment by ghsolomon** [comment]
> that is, we could just build this in to XH.reloadApp() and maybe avoid other caching issues as well.

Got it, yes that makes a lot of sense to me!

**Comment by lbwexler** [comment]
greg -- think this looks much better two questions:

+ Do we know that the standard reload wouldn't have worked?  Just occurred to me that this was not clear and just  want to understand this better.
+ Should we check that the cacheBuster flag date to make sure that we are not in a death loop?   (Am fine if the answer is no)

**Comment by ghsolomon** [comment]
> greg -- think this looks much better two questions:
> 
> * Do we know that the standard reload wouldn't have worked?  Just occurred to me that this was not clear and just  want to understand this better.
> * Should we check that the cacheBuster flag date to make sure that we are not in a death loop?   (Am fine if the answer is no)

I don't know that the standard reload wouldn't have worked.  This was belt-and-suspenders since the root issue had to do with the client holding onto an old version of the app for too long.  It might be unnecessary. Not sure the best way to test.

I don't think we _should_ end up in a death loop since `wasDiscarded` _shouldn't_ keep getting set to `true` after the first reload, but happy to put that check in to be safe.

**Pull request merged**